File: Bachelor/Semester4/Systems_for_Design_and_Implementation/Java/src/test/java/domain/validator/UserValidatorTest.java
package domain.validator;

import domain.User;
import junit.framework.TestCase;

public class UserValidatorTest extends TestCase {

    private final UserValidator validator = new UserValidator();

    public void testValidate() {
        User user = new User(1L,"abc","pass");
        try{
            validator.validate(user);
            fail();
        }catch (ValidationException e){
            assertEquals("Invalid password\n", e.getMessage());
        }
        user.setPassword("password");
        validator.validate(user);
        try{
            validator.validate(new User(2L,"",""));
            fail();
        }catch (ValidationException e){
            assertEquals("Invalid username\nInvalid password\n", e.getMessage());
        }
    }
}

File: Bachelor/Semester4/Systems_for_Design_and_Implementation/C#/ProiectCShMPP/Tests/domain/validator/UserValidatorTest.cs
using NUnit.Framework;
using ProiectCShMPP.domain;
using ProiectCShMPP.domain.validator;

namespace Tests.domain.validator
{
    [TestFixture]
    public class UserValidatorTest
    {
        private readonly IValidator<long, User> _validator = new UserValidator();
        
        [Test]
        public void TestValidate()
        {
            User user = new User(1L)
            {
                Password = "",
                Username = ""
            };
            try
            {
                _validator.Validate(user);
                Assert.Fail();
            }
            catch (ValidationException e)
            {
                Assert.AreEqual("Invalid username\nInvalid password\n",e.Message);
            }
            _validator.Validate(new User(2L)
            {
                Password = "notAVeryLongPassword",
                Username = "Username"
            });
        }
    }
}